# Summing warehouse businesses by zip code with ZBP data

DT

4/20/22

This requires the zip code data to be unzipped.
ZBP data is in the same folders as the CBPs.

In [6]:
import pandas as pd
import os

In [7]:
os.listdir('../raw_data/cbp19/')

['cbp19cd.xlsx',
 'cbp19co.zip',
 'cbp19csa.zip',
 'cbp19msa.zip',
 'cbp19pr_ia_co.zip',
 'cbp19pr_ia_st.zip',
 'cbp19st.zip',
 'cbp19us.zip',
 'zbp19detail.txt',
 'zbp19detail.zip',
 'zbp19totals.zip']

In [9]:
%%time
zbp19 = pd.read_csv('../raw_data/cbp19/zbp19detail.txt', encoding = 'ISO-8859-1')
zbp19.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2881311 entries, 0 to 2881310
Data columns (total 16 columns):
 #   Column    Dtype 
---  ------    ----- 
 0   zip       int64 
 1   name      object
 2   naics     object
 3   est       int64 
 4   n<5       object
 5   n5_9      object
 6   n10_19    object
 7   n20_49    object
 8   n50_99    object
 9   n100_249  object
 10  n250_499  object
 11  n500_999  object
 12  n1000     object
 13  city      object
 14  stabbr    object
 15  cty_name  object
dtypes: int64(2), object(14)
memory usage: 351.7+ MB
Wall time: 10.4 s


In [10]:
zbp19['zip'].value_counts()

33166    830
10001    816
92705    760
11101    746
80112    729
        ... 
72812      1
28272      1
72776      1
72769      1
501        1
Name: zip, Length: 35129, dtype: int64

In [13]:
zbp19['naics'].value_counts()[:20]

------    35129
81----    22504
23----    22427
44----    21916
238///    19475
72----    19076
813///    18897
722///    18117
62----    17955
541///    17748
54----    17748
72251/    16735
7225//    16735
56----    16539
813110    16416
8131//    16416
81311/    16416
48----    16118
561///    15812
31----    15775
Name: naics, dtype: int64

There are 35129 total businesses.
The big ones:
- 81----    is Other Services.
- 23----    is construction. (This could be useful when thinking about warehouses?? How do construction business work in terms of establishments vs sites?)
- 44----	Retail Trade
- 238///	Specialty Trade Contractors

More in the NAICS notebook.

It will be helpful to first just extract only the rows with zips in cali.